### ISSUE TITLE
Convenience method YearMonth.PlusMonths(x)?

### ISSUE THREAD

**Issue by ArneMancofi** [nodatime/nodatime#1536]
In version 3, if I want to find the `YearMonth` in two months' time, I know I can do
```csharp
new YearMonth(2020, 05).OnDayOfMonth(1).PlusMonths(2).ToYearMonth()
```
but that seems rather long-winded. Does a convenience method like
```csharp
new YearMonth(2020, 05).PlusMonths(2)
```
exist anywhere in Noda Time?

**Comment by jskeet** [comment]
No, this doesn't exist yet, but sounds reasonable. I don't expect to get to it before 3.0 though - you can always write your own extension method for now.

**Comment by carlosschults** [comment]
Can I give this a try, @jskeet ?

**Comment by jskeet** [comment]
Sure, that's fine :)

**Comment by carlosschults** [comment]
@jskeet OK, I need a little bit of help here. I'm just trying to run the tests—with `dotnet test`—before doing any development, and I'm getting some failed tests.

The first failing test is "AllDetectedNamesAreMaped":

`    System.TypeInitializationException : The type initializer for 'NodaTime.TzdbCompiler.Test.Tzdb.NameIdMappingSupportTest' threw an exception.
      ----> System.ArgumentException : An item with the same key has already been added. Key: Hora Padrão Austrália Ocidental
  Rastreamento de Pilha: 
    NameIdMappingSupportTest.AllDetectedNamesAreMapped() linha 34
    --ArgumentException
    Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
    Dictionary`2.Add(TKey key, TValue value)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
    NameIdMappingSupportTest.cctor() linha 16`

The second one is "AllDetectedNamesAreMappedCorrectly":

`System.TypeInitializationException : The type initializer for 'NodaTime.TzdbCompiler.Test.Tzdb.NameIdMappingSupportTest' threw an exception.
      ----> System.ArgumentException : An item with the same key has already been added. Key: Hora Padrão Austrália Ocidental
  Rastreamento de Pilha: 
    NameIdMappingSupportTest.AllDetectedNamesAreMappedCorrectly() linha 44
    --ArgumentException
    Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
    Dictionary`2.Add(TKey key, TValue value)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
    NameIdMappingSupportTest.cctor() linha 16`

I'm on Windows 10. The version of my .NET Core SDK is 3.1.100.

**Comment by jskeet** [comment]
I'll see if I can investigate over the weekend. Please let me know which culture you've got Windows set to.

**Comment by carlosschults** [comment]
Sure, sounds good.

My Windows is set to pt-BR (Brazilian Portuguese.)

Thanks!

**Comment by jskeet** [comment]
I've had a look at this, and it's going to be relatively tricky for me to either reproduce or fix. I suggest you just ignore that project entirely - just test NodaTime.Test and you should be fine.

**Comment by carlosschults** [comment]
Got it. Thanks for looking into it!
